### PR TITLE
ERC20 factory 

### PR DESCRIPTION
**Pull request by marceljay**
First version, only deployment, no permissions implemented.

I copied most of the code from the ERC20 implementation, as I expect a lot of custom functionality wrt minting/burning/retiring

To-Do (high prio)
- NFT receive functionality

### CODE REVIEW

File: packages/hardhat/contracts/PTokenFactory/ProjectToken.sol [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        ){
+        _name = name_;
+        _symbol = symbol_;
+        vintage = _vintage;
+        standard = _standard;
+        country = _country;
+    }
+
+
+    function totalSupply() public view virtual override returns (uint256) {
```
Comment (leolara) [inline packages/hardhat/contracts/PTokenFactory/ProjectToken.sol:45]: @marceljay I do not think these methods are necessary they are already implemented in openzeppelin

Comment (marceljay) [reply packages/hardhat/contracts/PTokenFactory/ProjectToken.sol:45]: yeah can change that, I just copied it from the ref implementation to get a better overview. Maybe this is not good style haha :D

Comment (jamespfarrell) [reply packages/hardhat/contracts/PTokenFactory/ProjectToken.sol:72]: @marceljay did you change the above?

**Comment by jamespfarrell** [comment]
@marceljay - what's the status of this PR?

**Comment by marceljay** [comment]
I would like to get it merged asap, I'm going to resolve the conflicts

**Comment by marceljay** [comment]
Closed because already merged via other branches

**Pull request closed without merging**